### PR TITLE
[network-data] add length checks to HandleCommissioningGet

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/core/thread/network_data_leader_ftd.cpp [hunk 1]
```diff
@@ -291,20 +291,22 @@ void Leader::HandleCommissioningGet(Coap::Message &aMessage, const Ip6::MessageI
...
+    while (offset + sizeof(tlv) <= aMessage.GetLength())
     {
         aMessage.Read(offset, sizeof(tlv), &tlv);
+        VerifyOrExit(tlv.GetLength() <= aMessage.GetLength() - offset - sizeof(tlv));
 
         if (tlv.GetType() == MeshCoP::Tlv::kGet)
         {
             length = tlv.GetLength();
-            aMessage.Read(offset + sizeof(MeshCoP::Tlv), length, tlvs);
+            aMessage.Read(offset + sizeof(tlv), length, tlvs);
```
Comment (abtink) [inline src/core/thread/network_data_leader_ftd.cpp]: Checking `length` is smaller than sizeof (tlvs)?

File: src/core/thread/network_data_leader_ftd.cpp [hunk 2]
```diff
@@ -291,20 +291,22 @@ void Leader::HandleCommissioningGet(Coap::Message &aMessage, const Ip6::MessageI
     uint8_t      tlvs[NetworkData::kMaxSize];
     uint8_t      length = 0;
 
-    while (offset < aMessage.GetLength())
+    while (offset + sizeof(tlv) <= aMessage.GetLength())
```
Comment (abtink) [inline src/core/thread/network_data_leader_ftd.cpp]: How about using `Tlv::GetValueOffset()` to find the value offset and length for TLV type `MeshCoP::Tlv::kGet` here?

Comment (jwhui) [reply src/core/thread/network_data_leader_ftd.cpp]: Just submitted #4194 to replace this PR.

**Comment by jwhui** [comment]
Replaced by #4194

**Pull request closed without merging**